Sources/ManageServer.php

```diff
@@ -1467,20 +1518,35 @@ function saveSettings(&$config_vars)
 		}
 		elseif (in_array($config_var, $config_bools))
 		{
-			if (!empty($_POST[$config_var]))
-				$new_settings[$config_var] = 1;
-			else
-				$new_settings[$config_var] = 0;
+			$new_settings[$config_var] = !empty($_POST[$config_var]);
+		}
+		elseif (isset($config_multis[$config_var]))
+		{
+			$is_acceptable_type = false;
+
+			foreach ($config_multis[$config_var] as $type)
+			{
+				$temp = $_POST[$config_var];
+				settype($temp, $type);
+
+				if ($temp == $_POST[$config_var])
+				{
+					$new_settings[$config_var] = $temp;
+					$is_acceptable_type = true;
+					break;
+				}
+			}
+
+			if (!$is_acceptable_type)
+				fatal_error('Invalid config_var \'' . $config_var . '\'');
```

Comment:
is error logging here intended?

Reply:
Honestly, I was just following the lead of the previous version of this code. Now that you mention it, yeah, we could simply log the error and drop the invalid value rather than failing so dramatically.

Reply:
On further consideration, we probably should leave this as a `fatal_error()`. If we just use `log_error()`, we will end up showing a "The settings were successfully saved" message to the admin even though not all the settings were saved. Unfortunately, the current architecture of these functions doesn't allow us to trigger the "Your changes were not saved because: ..." message from within `saveSettings()`, so a fatal error is the only way to give the admin truthful feedback at this time.

It should be possible to change things so that `saveSettings()` becomes able to cause the "Your changes were not saved because: ..." message to appear, but that would need to happen in a future PR.